mds/views.py

```diff
@@ -142,6 +130,27 @@ class DeviceViewSet(
     serializer_class = serializers.Device
     schema = CustomViewSchema()
 
+    def get_queryset(self):
+        queryset = models.Device.objects.prefetch_related(
+            Prefetch(
+                "telemetries",
+                queryset=models.Telemetry.objects.filter(
+                    id__in=Subquery(
+                        models.Telemetry.objects.filter(device_id=OuterRef("device_id"))
+                        .order_by("-timestamp")
+                        .values_list("id", flat=True)[:1]
+                    )
+                ),
+                to_attr="_latest_telemetry",
+            )
+        ).select_related("provider")
+
+        user = self.request.user
+        if isinstance(user, RemoteUser) and user.provider_id:
+            queryset = queryset.filter(provider_id=user.provider_id)
```

Comment:
```suggestion
    provider_id = getattr(self.request.user, "provider_id", None)
    if provider_id:
        queryset = queryset.filter(provider_id=provider_id)
```
This would avoid using `isinstance` (usually bad practice as it contradicts the duck typing principle)

Reply:
alright !